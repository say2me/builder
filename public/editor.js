/* eslint-disable import/no-webpack-loader-syntax */
import vcCake from 'vc-cake'

import 'public/tools/jqueryCaret' // used in tokenizationList.js
import publicAPI from './components/api/publicAPI'

import { start } from './components/editorInit/start'
import heartbeat from './components/heartbeat/index'
import { rebuildPosts } from './components/editorInit/rebuildPosts'

(() => {
  if (window.vcvPostUpdateAction && window.vcvPostUpdateAction === 'updatePosts') {
    rebuildPosts()
    return
  }
  let started = false
  const setStarted = () => {
    started = true
  }

  // Need to wait while ALL Elements will be initialized otherwise can break layout
  window.onload = () => {
    heartbeat(window.jQuery)
    start(setStarted)
  }
  // In case if jQuery.ready fails try to load manually
  window.setTimeout(() => {
    if (!started) {
      start(setStarted)
    }
  }, 10000)
})()

if (vcCake.env('VCV_DEBUG') === true) {
  window.app = vcCake
}
window.vc = publicAPI