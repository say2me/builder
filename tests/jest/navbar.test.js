/* global describe, test, expect */
import vcCake from 'vc-cake'
import '../../public/variables'
// Services & Storages
import '../../public/editor/services/utils/service.js'
import '../../public/editor/services/dataManager/service.js'
import '../../public/editor/services/document/service.js'
import '../../public/editor/services/hubElements/service.js'
import '../../public/editor/services/cook/service.js'
import '../../public/editor/services/modernAssetsStorage/service.js'
import '../../public/editor/services/api/service.js'
import '../../public/editor/services/wordpress-post-data/service'
import '../../public/config/wp-attributes'
import '../../public/editor/stores/elements/elementsStorage'
import '../../public/editor/stores/elements/elementSettings'
import '../../public/editor/modules/elementLimit/module'

// React and Enzyme
import React from 'react'
import renderer from 'react-test-renderer'
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'

// Navbar Component
import NavbarContainer from '../../public/components/navbar/navbarContainer'

configure({ adapter: new Adapter() });

describe('Tests editor navbar', () => {
  vcCake.env('platform', 'wordpress').start(() => {
    test('Create navbar component snapshot', () => {
      vcCake.add('contentLayout', (api) => {
        let navbarRef
        const navbar = renderer.create(
          <NavbarContainer getNavbarPosition={() => {}} wrapperRef={(navbar) => { navbarRef = navbar }} />
        )
        let tree = navbar.toJSON()
        expect(tree).toMatchSnapshot()
      })
    })
    test('Render navbar component', () => {
      let navbarRef
      const navbarContainer = mount(<NavbarContainer getNavbarPosition={() => {}} wrapperRef={(navbar) => { navbarRef = navbar }} />)
      expect(navbarContainer.exists()).toBe(true)
      const navbar = navbarContainer.find('.vcv-ui-navbar')
      expect(navbar.exists()).toBe(true)
      const addElementControl = navbar.find('.vcv-ui-navbar-control[data-vcv-guide-helper="plus-control"]')
      expect(addElementControl.exists()).toBe(true)
    })
  })
})