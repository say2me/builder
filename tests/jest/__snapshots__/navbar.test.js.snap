// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests editor navbar Create navbar component snapshot 1`] = `
<div
  className="vcv-layout-bar-header vcv-layout-bar-header--loading"
  id="vcv-editor-header"
>
  <div
    id="vc-navbar-container"
  >
    
    <div
      className="vcv-ui-navbar-container vcv-ui-navbar-is-detached"
    >
      <nav
        className="vcv-ui-navbar vcv-ui-navbar-hide-labels"
      >
        <div
          className="vcv-ui-navbar-drag-handler vcv-ui-drag-handler"
          onMouseDown={[Function]}
        >
          <i
            className="vcv-ui-drag-handler-icon vcv-ui-icon vcv-ui-icon-drag-dots"
          />
        </div>
        <a
          className="vcv-ui-navbar-logo"
          href="undefined"
          rel="noopener noreferrer"
          target="_blank"
          title="Visual Composer Website Builder"
        >
          <span
            className="vcv-ui-navbar-logo-title"
          >
            Visual Composer Website Builder
          </span>
        </a>
        <span
          className="vcv-ui-navbar-control"
          data-vcv-guide-helper="plus-control"
          onClick={[Function]}
          title="undefined (Shift + A)"
        >
          <span
            className="vcv-ui-navbar-control-content"
          >
            <i
              className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-add"
            />
            <span />
          </span>
        </span>
        <span
          className="vcv-ui-navbar-control"
          onClick={[Function]}
          title="undefined (Shift + T)"
        >
          <span
            className="vcv-ui-navbar-control-content"
          >
            <i
              className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-layers"
            />
            <span />
          </span>
        </span>
        <div
          className="vcv-ui-navbar-controls-group vcv-ui-navbar-sandwich--stop-close"
        >
          <span
            className="vcv-ui-navbar-control"
            disabled={true}
            onClick={[Function]}
            title="undefined (Ctrl + Z)"
          >
            <span
              className="vcv-ui-navbar-control-content"
            >
              <i
                className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-undo"
              />
              <span />
            </span>
          </span>
          <span
            className="vcv-ui-navbar-control"
            disabled={true}
            onClick={[Function]}
            title="undefined (Ctrl + Shift + Z)"
          >
            <span
              className="vcv-ui-navbar-control-content"
            >
              <i
                className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-redo"
              />
              <span />
            </span>
          </span>
        </div>
        <dl
          className="vcv-ui-navbar-dropdown vcv-ui-navbar-dropdown--layout vcv-ui-navbar-dropdown-linear vcv-ui-pull-end"
          data-vcv-guide-helper="layout-control"
          tabIndex="0"
        >
          <dt
            className="vcv-ui-navbar-dropdown-trigger vcv-ui-navbar-control"
            onClick={[Function]}
          >
            <span
              className="vcv-ui-navbar-control-content"
            >
              <i
                className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-desktop"
              />
              <span />
            </span>
          </dt>
          <dd
            className="vcv-ui-navbar-dropdown-content vcv-ui-navbar-dropdown-content--layout"
          >
            <span
              className="vcv-ui-navbar-control"
              onClick={[Function]}
            >
              <span
                className="vcv-ui-navbar-control-content"
              >
                <i
                  className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-desktop"
                />
                <span>
                  <em>
                    (Width: 1200 px)
                  </em>
                </span>
              </span>
            </span>
            <span
              className="vcv-ui-navbar-control"
              onClick={[Function]}
            >
              <span
                className="vcv-ui-navbar-control-content"
              >
                <i
                  className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-tablet-landscape"
                />
                <span>
                  <em>
                    (Width: 1220 px)
                  </em>
                </span>
              </span>
            </span>
            <span
              className="vcv-ui-navbar-control"
              onClick={[Function]}
            >
              <span
                className="vcv-ui-navbar-control-content"
              >
                <i
                  className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-tablet-portrait"
                />
                <span>
                  <em>
                    (Width: 818 px)
                  </em>
                </span>
              </span>
            </span>
            <span
              className="vcv-ui-navbar-control"
              onClick={[Function]}
            >
              <span
                className="vcv-ui-navbar-control-content"
              >
                <i
                  className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-mobile-landscape"
                />
                <span>
                  <em>
                    (Width: 664 px)
                  </em>
                </span>
              </span>
            </span>
            <span
              className="vcv-ui-navbar-control"
              onClick={[Function]}
            >
              <span
                className="vcv-ui-navbar-control-content"
              >
                <i
                  className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-mobile-portrait"
                />
                <span>
                  <em>
                    (Width: 340 px)
                  </em>
                </span>
              </span>
            </span>
            <span
              className="vcv-ui-navbar-control"
              onClick={[Function]}
            >
              <span
                className="vcv-ui-navbar-control-content"
              >
                <i
                  className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-multiple-devices"
                />
                <span>
                  <div
                    className="vcv-tooltip-container"
                  >
                    <i
                      className="vcv-tooltip-button vcv-ui-icon vcv-ui-icon-question vcv-ui-icon--light-hover"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                    />
                  </div>
                </span>
              </span>
            </span>
          </dd>
        </dl>
        <span
          className="vcv-ui-navbar-control vcv-ui-pull-end"
          data-vcv-guide-helper="hub-control"
          onClick={[Function]}
        >
          <span
            className="vcv-ui-navbar-control-content"
          >
            <i
              className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-hub-shop"
            />
            <span />
          </span>
        </span>
        <span
          className="vcv-ui-navbar-control vcv-ui-pull-end"
          data-vcv-guide-helper="settings-control"
          onClick={[Function]}
          title="undefined (Shift + S)"
        >
          <span
            className="vcv-ui-navbar-control-content"
          >
            <i
              className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-cog"
            />
            <span />
          </span>
        </span>
        <span
          className="vcv-ui-navbar-control-separator vcv-ui-pull-end"
        />
        <div
          className="vcv-ui-navbar-controls-group vcv-ui-pull-end"
          data-vcv-guide-helper="save-control"
        >
          <span
            className="vcv-ui-navbar-control"
            onClick={[Function]}
            title="undefined (Ctrl + S)"
          >
            <span
              className="vcv-ui-navbar-control-content"
            >
              <i
                className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-save"
              />
              <span />
            </span>
          </span>
        </div>
        <dl
          className="vcv-ui-navbar-dropdown vcv-ui-pull-end vcv-ui-navbar-sandwich"
        >
          <dt
            className="vcv-ui-navbar-dropdown-trigger vcv-ui-navbar-control"
            onClick={[Function]}
          >
            <span
              className="vcv-ui-navbar-control-content"
            >
              <i
                className="vcv-ui-navbar-control-icon vcv-ui-icon vcv-ui-icon-mobile-menu"
              />
              <span />
            </span>
          </dt>
          <dd
            className="vcv-ui-navbar-dropdown-content vcv-ui-navbar-show-labels"
          >
            <div
              style={
                Object {
                  "height": "auto",
                  "maxHeight": "80vh",
                  "minHeight": 0,
                  "overflow": "hidden",
                  "position": "relative",
                  "width": "100%",
                }
              }
            >
              <div
                className="vcv-ui-scroll-content"
                style={
                  Object {
                    "WebkitOverflowScrolling": "touch",
                    "bottom": undefined,
                    "left": undefined,
                    "marginBottom": 0,
                    "marginRight": 0,
                    "maxHeight": "calc(80vh + 0px)",
                    "minHeight": 0,
                    "overflow": "scroll",
                    "position": "relative",
                    "right": undefined,
                    "top": undefined,
                  }
                }
              >
                <div>
                  <div
                    className="vcv-ui-navbar-controls-set"
                  >
                    <span
                      className="vcv-ui-navbar-control"
                      onClick={[Function]}
                      title="undefined (Ctrl + Shift + P)"
                    >
                      <span
                        className="vcv-ui-navbar-control-content"
                      />
                    </span>
                    <span
                      className="vcv-ui-navbar-control"
                      onClick={[Function]}
                    >
                      <span
                        className="vcv-ui-navbar-control-content"
                      />
                    </span>
                    
                    <span
                      className="vcv-ui-navbar-control"
                      data-backend-editor="backendEditor"
                      onClick={[Function]}
                    >
                      <span
                        className="vcv-ui-navbar-control-content"
                      />
                    </span>
                  </div>
                </div>
              </div>
              <div
                className="vcv-ui-scroll-track--horizontal"
                style={
                  Object {
                    "display": "none",
                    "height": 6,
                    "position": "absolute",
                  }
                }
              >
                <div
                  className="vcv-ui-scroll-thumb--horizontal"
                  style={
                    Object {
                      "display": "block",
                      "height": "100%",
                      "position": "relative",
                    }
                  }
                />
              </div>
              <div
                className="vcv-ui-scroll-track--vertical"
                style={
                  Object {
                    "display": "none",
                    "position": "absolute",
                    "width": 6,
                  }
                }
              >
                <div
                  className="vcv-ui-scroll-thumb--vertical"
                  style={
                    Object {
                      "display": "block",
                      "position": "relative",
                      "width": "100%",
                    }
                  }
                />
              </div>
            </div>
          </dd>
        </dl>
        <div
          className="vcv-ui-navbar-drag-handler vcv-ui-navbar-controls-spacer"
          onMouseDown={[Function]}
        />
      </nav>
    </div>
  </div>
</div>
`;
